spring-ai-client-chat/src/main/java/org/springframework/ai/chat/client/advisor/MessageChatMemoryAdvisor.java

```diff
@@ -95,12 +90,39 @@ private ChatClientRequest before(ChatClientRequest chatClientRequest) {
 
 		// 4. Add the new user message to the conversation memory.
 		UserMessage userMessage = processedChatClientRequest.prompt().getUserMessage();
-		this.getChatMemoryStore().add(conversationId, userMessage);
+		this.chatMemory.add(conversationId, userMessage);
 
 		return processedChatClientRequest;
 	}
 
-	private void after(ChatClientResponse chatClientResponse) {
+	protected String doGetConversationId(Map<String, Object> context) {
```

Comment:
This can be removed in favour of `getConversationId()` provided by `BaseChatMemoryAdvisor`.